src/landing_game_object.py

```diff
@@ -19,6 +22,8 @@ def __init__(
             pos (Vec2d, optional): 2d Position in pixel. (0,0) represents left top corner. Defaults to Vec2d().
         """
         super().__init__()
+        self.ID = self.ID_generator.assign_ID()
+        self._name: str = None
```

Comment:
As discussed, do not store redundant name and I'd in parallel

Reply:
Removed names completely